sap-2-tier-marketplace-image/shared/os-disk-parts.json

```diff
@@ -100,7 +101,7 @@
             "uri": "[parameters('userImageVhdUri')]"
           },
           "vhd": {
-            "uri": "[concat('http://',parameters('userImageStorageAccount'),'.blob.core.windows.net/', resourceGroup().name, '/', parameters('vmName'), '-', variables('osDiskName'))]"
+            "uri": "[concat('http://',parameters('userImageStorageAccount'),'.blob.core.windows.net/', variables('resourceGroupLower'), '/', parameters('vmName'), '-', variables('osDiskName'))]"
```

Comment:
You're using toLower(resourceGroup().name) for the name of the storage container.  The resource group name can contain an underscore.  The container name cannot.  This template will fail to deploy when the resource group name contains an underscore.